device/tests/Microsoft.Azure.Devices.Client.Test/ModuleAuthenticationWithTokenRefreshTests.cs

```diff
@@ -0,0 +1,260 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information
+
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Diagnostics;
+using Microsoft.Azure.Amqp;
+
+namespace Microsoft.Azure.Devices.Client.Test
+{
+    [TestClass]
+    public class ModuleAuthenticationWithTokenRefreshTests
+    {
+        private const string TestDeviceId = "TestDeviceID";
+        private const string TestModuleId = "TestModuleID";
+        private const string TestIoTHubName = "contoso.azure-devices.net";
+        private const int DefaultTimeToLiveSeconds = 1 * 60 * 60;
+        private static string TestSharedAccessKey;
+
+        static ModuleAuthenticationWithTokenRefreshTests()
+        {
+            var rnd = new Random();
+            var rndBytes = new byte[32];
+            rnd.NextBytes(rndBytes);
+
+            TestSharedAccessKey = Convert.ToBase64String(rndBytes);
+        }
+
+        [TestMethod]
+        public void ModuleAuthenticationWithTokenRefresh_Ctor_WrongArguments_Fail()
```

Comment:
If we extract a common abstract, we should unify testing against that (a single copy of the tests) and only test differences between `Device` and `Method`.